docs/docsite/rst/user_guide/playbooks_strategies.rst

```diff
@@ -154,17 +156,18 @@ Ordering execution based on inventory
 The ``order`` keyword controls the order in which hosts are run. Possible values for order are:
 
 inventory:
-    (default) The order provided in the inventory
+    (default) The order provided by the inventory for the selection requested (see note below)
 reverse_inventory:
-    The reverse of the order provided by the inventory
+    The same as above, but reversing the returned list
 sorted:
     Sorted alphabetically sorted by name
 reverse_sorted:
     Sorted by name in reverse alphabetical order
 shuffle:
     Randomly ordered on each run
 
-Other keywords that affect play execution include ``ignore_errors``, ``ignore_unreachable``, and ``any_errors_fatal``. These options are documented in :ref:`playbooks_error_handling`.
+.. note::
+    The 'inventory' order does not equate to the order in which hosts/groups are defined in the inventory source file, but the 'order in which a selection is returned from the compiled inventory'. This is a backwards compatible option and while reproducible it is not normally predictable. Due to the nature of inventory, host patterns, limits, inventory plugins and the ability to allow multiple sources it is almost impossible to return such an order. While for simple cases this might happen to match the file definition order, it is not guaranteed.
```

Comment:
```suggestion
    The 'inventory' order does not equate to the order in which hosts/groups are defined in the inventory source file, but the 'order in which a selection is returned from the compiled inventory'. This is a backwards compatible option and while reproducible it is not normally predictable. Due to the nature of inventory, host patterns, limits, inventory plugins, and the ability to allow multiple sources, it is almost impossible to return such an order. While for simple cases this might happen to match the file definition order, it is not guaranteed.
```